lib/spack/llnl/util/filesystem.py

```diff
@@ -568,20 +568,22 @@ def find_libraries(args, root, shared=True, recurse=False):
     """Returns an iterable object containing a list of full paths to
     libraries if found.
 
-    Args:
-        args: iterable object containing a list of library names to \
-            search for (e.g. 'libhdf5')
-        root: root folder where to start searching
-        shared: if True searches for shared libraries, otherwise for static
-        recurse: if False search only root folder, if True descends top-down \
-            from the root
-
-    Returns:
-        list of full paths to the libraries that have been found
+    :param args: Library name(s) to search for
+    :type args: str or collections.Sequence
+    :param str root: The root directory to start searching from
+    :param bool shared: if True searches for shared libraries,
+        otherwise for static
+    :param bool recurse: if False search only root folder,
+        if True descends top-down from the root
+
+    :returns: The libraries that have been found
+    :rtype: LibraryList
     """
-    if not isinstance(args, collections.Sequence) or isinstance(args, str):
```

Comment:
No, this is just to rule out strings, because they are a sequence:
```python
>>> import collections
>>> isinstance([1, 2, 3], collections.Sequence)
True
>>> isinstance('abc', collections.Sequence)
True
```
but we don't want to search for `a`, `b` and `c` separately.